optuna/samplers/_partial_fixed.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any
+from typing import Dict
+import warnings
+
+import numpy
+
+from optuna.distributions import BaseDistribution
+from optuna.samplers import BaseSampler
+from optuna.study import Study
+from optuna.trial import FrozenTrial
+
+
+class PartialFixedSampler(BaseSampler):
+    """Sampler that can sample with some parameters fixed.
+
+        .. versionadded:: 2.2.0
+
+    Example:
+
+        After optimizing with :class:`~optuna.samplers.TPESampler`,
+        fix the value of ``y`` and optimize again with :class:`~optuna.samplers.CmaEsSampler`.
+
+        .. testcode::
+
+            import optuna
+
+            def objective(trial):
+                x = trial.suggest_uniform("x", -1, 1)
+                y = trial.suggest_int("y", -1, 1)
+                return x ** 2 + y
+
+            study = optuna.create_study()
+            study.optimize(objective, n_trials=10)
+
+            best_params = study.best_params
+            fixed_params = {"y": best_params["y"]}
+            base_sampler = optuna.samplers.RandomSampler()
+            partial_sampler = PartialFixedSampler(fixed_params, base_sampler)
+
+            study.sampler = partial_sampler
+            study.optimize(objective, n_trials=10)
```

Comment:
I think the "changing sampler of a study from TPE to CMA-ES" is not the main purpose to introduce the `PartialFixedSampler`. We can simplify the example.

```suggestion
        After several steps of optimization, you can fix the value of ``y`` and re-optimize it.

        .. testcode::

            import optuna

            def objective(trial):
                x = trial.suggest_uniform("x", -1, 1)
                y = trial.suggest_int("y", -1, 1)
                return x ** 2 + y

            study = optuna.create_study()
            study.optimize(objective, n_trials=10)

            best_params = study.best_params
            fixed_params = {"y": best_params["y"]}
            partial_sampler = PartialFixedSampler(fixed_params, study.sampler)

            study.sampler = partial_sampler
            study.optimize(objective, n_trials=10)
```